README.md

```diff
@@ -34,10 +43,117 @@ AWS S3
  - username: `longkey`
  - password: `verysecretkey`
  
- ## Running tests locally
+### Running tests
 
 * `make tests` runs unit tests.
 
+## Documentation
+
+### Architecture
+
+This project uses [Apache Airflow](https://airflow.apache.org/) and has been
+implemented according to the official documentation. Please refer
+to the official documentation for detailed information about Apache Airflow.
+
+In short, Apache Airflow is comprised of six main components: a web server, 
+a scheduler, workers, a message broker, DAGs and a database.
+
+The **Web Server** provides the interface to Apache Airflow at run time. This 
+can done using either the [web interface](https://airflow.apache.org/ui.html) 
+or [HTTP endpoints](https://airflow.apache.org/api.html).
+
+The [Scheduler](https://airflow.apache.org/scheduler.html) is the coordinator of
+tasks.
+
+The [Workers](https://docs.celeryproject.org/en/latest/userguide/workers.html) 
+are processes available to perform tasks queued by the scheduler. 
+
+The **Message Broker** is the mechanism used for communication between the 
+scheduler and the workers.
+
+A [DAG](https://airflow.apache.org/concepts.html#dags) is a python file with a
+series of functions or classes in the form of 
+[Operators](https://airflow.apache.org/concepts.html#operators) 
+that represent tasks. Each task is then executed according to the 
+[DAG composition](https://airflow.apache.org/concepts.html#bitshift-composition).
+
+![Simple Apache Airflow Architectural Diagram](https://miro.medium.com/max/1140/1*u6duhZD2J_i1zZ0Txq26Cg.png)
+
+### DAGs
+Airflow inspects the `dags/` folder for files that instantiate the DAG class.
+Airflow expects to find this directory in the `AIRFLOW_HOME` directory.
+
+### Configuration
+Configuration files can be found in the `config/` directory. Airflow expects to 
+find a file with the name `airflow.cfg` in the `AIRFLOW_HOME` directory. Libero
+specific have been added under the heading of `[libero]`. These can then be read
+using the following:
+```python
+from airflow import configuration
+
+SEARCH_URL = configuration.conf.get('libero', 'search_url')
+```
+
+### Tests
+Tests are run using [pytest](https://pytest.org/en/latest/). Test files are 
+located in the `tests/` directory. Files needed for testing, such as archives and 
+xml files, are placed in the `tests/assets/` directory. Before tests are run an
+sqlite database is used and initialized as some operations require database
+access.
+ 
+Some operators such as the `PythonOperator` expect a callable to be parsed to 
```

Comment:
can you add an example of another Operator (doesn't matter which one)?

Reply:
Why?

Reply:
This section is in reference to the context object during testing.

Reply:
We don't use other Operators I guess, so fine as it is.